### PR TITLE
Extended Sharing Functionalities

### PR DESCRIPTION
**Pull request by wyatt-howe**
- Convenient functions for sharing common data structures, as well as serialization and opening

### CODE REVIEW

**Comment by wyatt-howe** [comment]
This seems stable now, with the exception of a rare op_id or share_id issue.

**Comment by p-flock** [comment]
Can you add tests similar to the way we test regular sharing? Just confirming that what you open is the same as what you shared etc. The same structure as the base: share testing.

File: tests/suite/config/array/computations-share.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    return values['numbers'][jiff_instance.id];
+  }
+
+  // MPC interpreter
+  // If we got here, we must be either a sender or a receiver
+
+  // Case 1: did not receive any shares, must have sent one, will receive an open.
+  if (shareParameters.receivers.indexOf(jiff_instance.id) === -1) {
+    console.log('did not receive any shares:', shareParameters)
+    return jiff_instance.receive_open(shareParameters.receivers, shareParameters.threshold);
```
Comment (p-flock) [inline tests/suite/config/array/computations-share.js]: This should be `jiff_instance.receive_nd_open()` as jiff needs to know it's expecting an array not just a single share

**Comment by p-flock** [comment]
Work on this feature is now on [this branch](https://github.com/multiparty/jiff/tree/array_sharing). The refactor made a merge cumbersome, so code for the array sharing functionalities was cherry picked from here and imported into the new strucure.

**Pull request closed without merging**